roles/dns_client/tasks/main.yml

```diff
@@ -0,0 +1,17 @@
+---
+- name: Get OpenShift facts
+  openshift_facts:
+    local_facts:
+      hostname: "{{ openshift_hostname | default(None) }}"
+      ip: "{{ openshift_ip | default(None) }}"
+      public_hostname: "{{ openshift_public_hostname | default(None) }}"
+      public_ip: "{{ openshift_public_ip | default(None) }}"
+
+- name: Configure the resolver
+  template: dest=/etc/resolv.conf src=resolv.conf
```

Comment:
We should also be modifying `/etc/dhcp/dhclient-{{ ansible_default_ipv4.interface }}.conf` to add `prepend domain-name-servers {{ dns_server }}` for hosts configured to use dhcp instead of static IPs. 

This way we still accept the upstream dns servers, they are just used as alternatives to the one that we create.  Also, we should not be configuring a recursive resolver, so this will allow us to fail back to upstream DNS for domains we do not specify.